core/server/master/src/main/java/alluxio/master/file/contexts/CreatePathContext.java

```diff
@@ -81,13 +82,17 @@ protected CreatePathContext(T optionsBuilder) {
     }
     // Initialize mPersisted based on proto write type.
     WritePType writeType = WritePType.NONE;
+    mXAttr = null;
     if (optionsBuilder instanceof CreateFilePOptions.Builder) {
       writeType = ((CreateFilePOptions.Builder) optionsBuilder).getWriteType();
+      mXAttr = CommonUtils.convertFromByteString(
+          ((CreateFilePOptions.Builder) optionsBuilder).getXattrMap());
     } else if (optionsBuilder instanceof CreateDirectoryPOptions.Builder) {
       writeType = ((CreateDirectoryPOptions.Builder) optionsBuilder).getWriteType();
+      mXAttr = CommonUtils.convertFromByteString(
+          ((CreateDirectoryPOptions.Builder) optionsBuilder).getXattrMap());
```

Comment:
This functionality for setting `xAttr` with `CreateDirectoryPOptions` is actually unused in this PR since the S3 API does not support simultaneous bucket creation and tagging.

I decided to include the change anyway for completeness of making `xAttr` configurable via Alluxio RPCs.